<style>
body h1:before, body h2:before, body h3:before, body h4:before, body h5:before, body h6:before {
  content: "";
  counter-increment: none;
}
</style>

# Week05.2 Team Activity
Due Saturday May 6th, 2023 by 11:59pm - please upload PDF with photograph (either rendered or a separate file) 

If you missed class – no credit for this activity.  

- **Please identify your team & the names of your teammates.** 

- **Please insert a team photo somewhere in this notebook.  If you cannot get the to render correctly to PDF please upload a separate file of the photo**

- **Please make certain that all of your teammates agree upon and understand the team answers. Thank you.**



## Stats 21 Teams 

We can use a little more practice with grouping/aggregation and sorting.

New data - Crimes for West Los Angeles from 1/1/2020 to 5/3/2023

I will do the reading and some prep for you

In [1]:
import pandas as pd

crime = pd.read_csv('WestLA_Crime_2020.csv', index_col = 'DR_NO')
crime.head()

Date Rptd                DATE OCC  TIME OCC  AREA   
DR_NO                                                                       
200804086  01/03/2020 12:00:00 AM  01/01/2020 12:00:00 AM      2111     8  \
200804017  01/01/2020 12:00:00 AM  01/01/2020 12:00:00 AM       100     8   
200804315  01/09/2020 12:00:00 AM  01/01/2020 12:00:00 AM       930     8   
200807697  03/25/2020 12:00:00 AM  01/01/2020 12:00:00 AM         1     8   
230808643  04/25/2023 12:00:00 AM  01/01/2020 12:00:00 AM       800     8   

          AREA NAME  Rpt Dist No  Part 1-2  Crm Cd   
DR_NO                                                
200804086   West LA          831         1     343  \
200804017   West LA          897         1     330   
200804315   West LA          828         1     310   
200807697   West LA          827         2     956   
230808643   West LA          801         1     341   

                                                 Crm Cd Desc   
DR_NO                                                          
200804086           SHOPLIFTING-GRAND THEFT ($950.01 & OVER)  \
200804017                              BURGLARY FROM VEHICLE   
200804315                                           BURGLARY   
200807697            LETTERS, LEWD  -  TELEPHONE CALLS, LEWD   
230808643  THEFT-GRAND ($950.01 & OVER)EXCPT,GUNS,FOWL,LI...   

                                 Mocodes  ...  Status   Status Desc Crm Cd 1   
DR_NO                                     ...                                  
200804086            1822 1420 0326 0325  ...      IC   Invest Cont    343.0  \
200804017  0358 1302 1307 1609 0344 1822  ...      IC   Invest Cont    330.0   
200804315            0344 1414 1822 1403  ...      IC   Invest Cont    310.0   
200807697                 1906 1913 1258  ...      AA  Adult Arrest    956.0   
230808643                 0352 1822 0344  ...      IC   Invest Cont    341.0   

           Crm Cd 2 Crm Cd 3  Crm Cd 4   
DR_NO                                    
200804086       NaN      NaN       NaN  \
200804017       NaN      NaN       NaN   
200804315       NaN      NaN       NaN   
200807697       NaN      NaN       NaN   
230808643       NaN      NaN       NaN   

                                           LOCATION Cross Street      LAT   
DR_NO                                                                       
200804086  12000    WILSHIRE                     BL          NaN  34.0473  \
200804017   3300    CASTLE HEIGHTS               AV          NaN  34.0314   
200804315  10800    WILSHIRE                     BL          NaN  34.0649   
200807697    500    LORNA                        LN          NaN  34.0557   
230808643  16700    MONTE HERMOSO                DR          NaN  34.0677   

                LON  
DR_NO                
200804086 -118.4768  
200804017 -118.3998  
200804315 -118.4372  
200807697 -118.4685  
230808643 -118.5520  

[5 rows x 27 columns]

In [2]:
crime.columns

Index(['Date Rptd', 'DATE OCC', 'TIME OCC', 'AREA', 'AREA NAME', 'Rpt Dist No',
       'Part 1-2', 'Crm Cd', 'Crm Cd Desc', 'Mocodes', 'Vict Age', 'Vict Sex',
       'Vict Descent', 'Premis Cd', 'Premis Desc', 'Weapon Used Cd',
       'Weapon Desc', 'Status', 'Status Desc', 'Crm Cd 1', 'Crm Cd 2',
       'Crm Cd 3', 'Crm Cd 4', 'LOCATION', 'Cross Street', 'LAT', 'LON'],
      dtype='object')

In [3]:
crime.dtypes

Date Rptd          object
DATE OCC           object
TIME OCC            int64
AREA                int64
AREA NAME          object
Rpt Dist No         int64
Part 1-2            int64
Crm Cd              int64
Crm Cd Desc        object
Mocodes            object
Vict Age            int64
Vict Sex           object
Vict Descent       object
Premis Cd           int64
Premis Desc        object
Weapon Used Cd    float64
Weapon Desc        object
Status             object
Status Desc        object
Crm Cd 1          float64
Crm Cd 2          float64
Crm Cd 3          float64
Crm Cd 4          float64
LOCATION           object
Cross Street       object
LAT               float64
LON               float64
dtype: object

In [4]:
# Convert datetime column to Pandas datetime format
crime['datetimeocc'] = pd.to_datetime(crime['DATE OCC'], format='%m/%d/%Y %I:%M:%S %p')
crime['year'] = crime['datetimeocc'].dt.strftime('%Y')
crime['month'] = crime['datetimeocc'].dt.strftime('%B')
crime['day'] = crime['datetimeocc'].dt.strftime('%d')
crime['YM'] = crime['datetimeocc'].dt.to_period('M')


In [5]:
crime.head()

Date Rptd                DATE OCC  TIME OCC  AREA   
DR_NO                                                                       
200804086  01/03/2020 12:00:00 AM  01/01/2020 12:00:00 AM      2111     8  \
200804017  01/01/2020 12:00:00 AM  01/01/2020 12:00:00 AM       100     8   
200804315  01/09/2020 12:00:00 AM  01/01/2020 12:00:00 AM       930     8   
200807697  03/25/2020 12:00:00 AM  01/01/2020 12:00:00 AM         1     8   
230808643  04/25/2023 12:00:00 AM  01/01/2020 12:00:00 AM       800     8   

          AREA NAME  Rpt Dist No  Part 1-2  Crm Cd   
DR_NO                                                
200804086   West LA          831         1     343  \
200804017   West LA          897         1     330   
200804315   West LA          828         1     310   
200807697   West LA          827         2     956   
230808643   West LA          801         1     341   

                                                 Crm Cd Desc   
DR_NO                                                          
200804086           SHOPLIFTING-GRAND THEFT ($950.01 & OVER)  \
200804017                              BURGLARY FROM VEHICLE   
200804315                                           BURGLARY   
200807697            LETTERS, LEWD  -  TELEPHONE CALLS, LEWD   
230808643  THEFT-GRAND ($950.01 & OVER)EXCPT,GUNS,FOWL,LI...   

                                 Mocodes  ...  Crm Cd 4   
DR_NO                                     ...             
200804086            1822 1420 0326 0325  ...       NaN  \
200804017  0358 1302 1307 1609 0344 1822  ...       NaN   
200804315            0344 1414 1822 1403  ...       NaN   
200807697                 1906 1913 1258  ...       NaN   
230808643                 0352 1822 0344  ...       NaN   

                                           LOCATION Cross Street      LAT   
DR_NO                                                                       
200804086  12000    WILSHIRE                     BL          NaN  34.0473  \
200804017   3300    CASTLE HEIGHTS               AV          NaN  34.0314   
200804315  10800    WILSHIRE                     BL          NaN  34.0649   
200807697    500    LORNA                        LN          NaN  34.0557   
230808643  16700    MONTE HERMOSO                DR          NaN  34.0677   

                LON  datetimeocc  year    month day       YM  
DR_NO                                                         
200804086 -118.4768   2020-01-01  2020  January  01  2020-01  
200804017 -118.3998   2020-01-01  2020  January  01  2020-01  
200804315 -118.4372   2020-01-01  2020  January  01  2020-01  
200807697 -118.4685   2020-01-01  2020  January  01  2020-01  
230808643 -118.5520   2020-01-01  2020  January  01  2020-01  

[5 rows x 32 columns]

## Try the following

The Westwood area is comprised of Rpt Dist No in 818, 817, 816, 837, 838, 828 (campus, Westwood Village, Fraternity Row, West of Campus, Sorority Row). 

1. Create a subset of data for the Westwood area - that is all the programming for today.

2. If you were interested in determining the mean age of crime victims (Vict Age) for the Westwood area by Month, would it be more appropriate to use a groupby() or value_count() or it doesn't matter (can do it both ways)

3. If you were interested in determining the number of crimes reported by Month starting with 1/2020 would it be more appropriate to use a groupby() or value_count() or it doesn't matter (can do it both ways)

(optional) See if your team can program solutions to 2 and 3

Resources that might help

https://data.lacity.org/Public-Safety/Crime-Data-from-2020-to-Present/2nrs-mtv8

https://lapdonlinestrgeacc.blob.core.usgovcloudapi.net/lapdonlinemedia/2022/01/WLA11x17.pdf

As always, what is important is a team effort and that everyone on the team has an idea of how to program these things and also feels free to ask the team for more understanding.  

Because we are in the process of forming final teams, if you are having trouble coming together as a team to complete an assignment like this one,  you should look for a different team for the next activity.

## What to submit for credit

- Please upload the Jupyter Notebook (.ipynb) and PDF (preferred) OR .html (OK) to Bruin Learn with your team attempt.  Please make sure every team member understands the contents of the notebook.  

- This is due before 11:59pm Saturday 5/6/2023 